src/service/variable-source.js

```diff
@@ -121,6 +121,9 @@ export class VariableSource {
 
     /** @private {boolean} */
     this.initialized_ = false;
+
+    /** @const @private {!Array<string>|undefined} */
```

Comment:
If this is private, and if the type checker is doing the correct thing, then subclasses should never be able to access this field... I think we really want a (ideally-abstract) method.

That said, why do we implement the actual meta tag check in the base class instead of here in the base class? Are there other implementations of `VariableSource` that makes whitelisting using meta tags not make sense?